infrastructure/Taskfile.yml

```diff
@@ -897,6 +897,26 @@ tasks:
           fi
       preconditions:
       - *require_site
+      - &require_clean_state {
```

Comment:
I am really not sure aboyt the [] encapsulation, but if you say it works as a multiline delimiter I trust you :)
I think there are problems with quoting and a small typo:
```bash
  current_branch=$(git branch --show-current)
  if [ "$current_branch" != "main" ]; then
    echo "Can't deploy if not on main branch"
    exit 1
  fi

  if ! git diff-index --quiet HEAD --; then
    echo "Cant deploy if there are changes on the branch"
    exit 1
  fi

  exit 0